### PR TITLE
MINOR:Type Casting Correction AND Null Pointer Exception (NPE) Defense

### PR DESCRIPTION
**Pull request by highluck**
MINOR: remove unnecessary code AND Type clarification

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by highluck** [comment]
ping @chia7712

**Comment by highluck** [comment]
ping @chia7712 
thanks!

**Comment by highluck** [comment]
@mimaison 

Can you review this? thanks!

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java [hunk 1]
```diff
@@ -437,7 +438,10 @@ private void rewriteSingleStoreSelfJoin(
         if (currentNode instanceof StreamStreamJoinNode && currentNode.parentNodes().size() == 1) {
             final StreamStreamJoinNode joinNode = (StreamStreamJoinNode) currentNode;
             // Remove JoinOtherWindowed node
-            final GraphNode parent = joinNode.parentNodes().stream().findFirst().get();
+            final GraphNode parent = joinNode.parentNodes().stream()
```
Comment (mimaison) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java:441]: It seems we checked there is one item in parentNodes in the `if` condition just above, so do we really need this?

Comment (highluck) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java:441]: As you said, it doesn't seem necessary.

**Comment by mjsax** [comment]
Closing this stale PR -- if it's still relevant, please re-open.

**Pull request closed without merging**